server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -548,10 +549,26 @@ protected Node(
              */
             this.environment = new Environment(settings, initialEnvironment.configDir(), Node.NODE_LOCAL_STORAGE_SETTING.get(settings));
             Environment.assertEquivalent(initialEnvironment, this.environment);
+            Stream<IndexStoreListener> indexStoreListenerStream = pluginsService.filterPlugins(IndexStorePlugin.class)
```

Comment:
Please move it under `if (DiscoveryNode.isSearchNode(settings) == false) {`, no need to create it otherwise